Add option to change directory before starting worker.

benprew opened the pull request:
This is used when your deploy system uses symlinks to manage
releases (ie Capistrano).  If you do a reload after a deploy, the
workers are still in the old deploy directory, so they won't pick up
the new code changes.

Commit:
Add option to change directory before starting worker.

This is used when your deploy system uses symlinks to manage
releases (ie Capistrano).  If you do a reload after a deploy, the
workers are still in the old deploy directory, so they won't pick up
the new code changes.

kmullin commented:
So, I am thinking this should actually be a default that chdirs this every time it forks. This will ensure new workers start in the designated working dir, transparently... thoughts?

I cant immediately think of a reason why this would need to be optional.

benprew commented:
Kevin,

That's a good point. I changed the commit to do the chdir automatically and removed it as a parameter.  And I like not having to add another option to the script.

Thanks!

benprew commented:
In testing this change, we've run into another issue.  Capistrano will remove old release directories and the parent kewatcher process will eventually not have a directory anymore, and then it starts throwing errors (see below).

So, I'm going to close this pull request, since reload doesn't work well with capistrano.

 Apr 12 16:00:07 StagingWorker kewatcher: /usr/lib64/ruby/gems/2.0.0/gems/bundler-1.5.3/lib/bundler/definition.rb:23:in `build': /var/kong/analytics_pipeline/releases/20140408203744/Gemfile not found (Bundler::GemfileNotFound)
Apr 12 16:00:07 StagingWorker kewatcher: from /usr/lib64/ruby/gems/2.0.0/gems/bundler-1.5.3/lib/bundler.rb:152:in`definition'
Apr 12 16:00:07 StagingWorker kewatcher: from /usr/lib64/ruby/gems/2.0.0/gems/bundler-1.5.3/lib/bundler.rb:115:in `setup'
Apr 12 16:00:07 StagingWorker kewatcher: from /usr/lib64/ruby/gems/2.0.0/gems/bundler-1.5.3/lib/bundler/setup.rb:17:in`<top (required)>'
Apr 12 16:00:07 StagingWorker kewatcher: from /usr/lib64/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:53:in `require'
Apr 12 16:00:07 StagingWorker kewatcher: from /usr/lib64/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:53:in`require'
